gensim/models/word2vec.py

```diff
@@ -1185,6 +1192,89 @@ def most_similar(self, positive=[], negative=[], topn=10, restrict_vocab=None):
         result = [(self.index2word[sim], float(dists[sim])) for sim in best if sim not in all_words]
         return result[:topn]
 
+    def wmdistance(self, document1, document2, WCD=False, RWMD=False):
+        """
+        Compute the Word Mover's Distance between two documents. When using this
+        code, please consider citing the following papers:
+        * Ofir Pele and Michael Werman, "A linear time histogram metric for improved SIFT matching".
+        * Ofir Pele and Michael Werman, "Fast and robust earth mover's distances".
+        * Matt Kusner et al. "From Word Embeddings To Document Distances".
+
+        Note that if one of the documents have no words that exist in the
+        Word2Vec vocab, `float('inf')` (i.e. infinity) will be returned.
+
+        This method only works if the C extension for `pyemd` is loaded (Gensim
+        should be installed with a working C compiler).
+
+        Example:
+        > # Train word2vec model.
+        > model = Word2Vec(sentences)
+
+        > # Some sentences to test.
+        > sentence1 = 'Obama speaks to the media in Illinois'.lower().split()
+        > sentence2 = 'The president greets the press in Chicago'.lower().split()
+
+        > # Remove their stopwords.
+        > from nltk.corpus import stopwords
+        > stopwords = nltk.corpus.stopwords.words('english')
+        > sentence1 = [w for w in sentence1 if w not in stopwords]
+        > sentence2 = [w for w in sentence2 if w not in stopwords]
+
+        > # Compute WMD.
+        > distance = model.wmdistance(sentence1, sentence2)
+        """
+
+        # TODO: is this the proper way of handling this?
```

Comment:
raise ImportError("pyemd package is required to do this. Please install using "pip install pyemd".")